### ISSUE TITLE
Content-Length header not sent for requests larger than 10 kB

### ISSUE THREAD

**Issue by augi** [http4s/http4s#2363]
Due to [this constant](https://github.com/http4s/http4s/blob/master/blaze-core/src/main/scala/org/http4s/blazecore/util/CachingChunkWriter.scala#L17) in private class, when making a request larger than 10 kB, the `Content-Length` header is not sent (because that buffer is flushed and so headers must be also written).

It would be nice to make this constant configurable, and maybe make the default value little bit higher (e.g. 1 MB).

**Comment by augi** [comment]
Yep, I will try to prepare a PR.

**Comment by augi** [comment]
I tried to prepare a PR but I'm not successful :( I'm not able to handle so large project with `sbt` and track what must be changed.

My attempt is [in this commit](https://github.com/augi/http4s/commit/0279b70eff0bca7b8aad0234ce64e39f280d7216).

1. I removed default value from the `bufferMaxSize` parameter of `CachingChunkWriter`.
2. It's used in `Http1Stage` so I introduces a new _abstract val_ named `chunkBufferMaxSize`.
3. I wasn't able to track what I should change :(
